annlite/core/index/hnsw/index.py

```diff
@@ -81,43 +61,36 @@ def __init__(
         self.ef_search = ef_search
         self.max_connection = max_connection
         self.pq_codec = pq_codec
-        self._set_backend_pq = False
         self.index_file = index_file
 
         self._init_hnsw_index()
 
     def _init_hnsw_index(self):
         self._index = Index(space=self.space_name, dim=self.dim)
-        if self.index_file and os.path.exists(self.index_file):
-            logger.info(
-                f'indexer will be loaded from {self.index_file}',
-            )
-            self.load(self.index_file)
-        else:
-            if self.index_file:
-                raise FileNotFoundError(
-                    f'index path: {self.index_file} does not exist',
+        if self.index_file:
+            if os.path.exists(self.index_file):
+                logger.info(
+                    f'indexer will be loaded from {self.index_file}',
                 )
-            if self.pq_codec is not None and self.pq_codec.is_trained:
-                self._index.init_index(
-                    max_elements=self.capacity,
-                    ef_construction=self.ef_construction,
-                    M=self.max_connection,
-                    pq_codec=self.pq_codec,
-                )
-                self._set_backend_pq = True
+                self.load(self.index_file)
             else:
-                self._index.init_index(
-                    max_elements=self.capacity,
-                    ef_construction=self.ef_construction,
-                    M=self.max_connection,
-                    pq_codec=None,
+                raise FileNotFoundError(
+                    f'index path: {self.index_file} does not exist',
                 )
-                self._set_backend_pq = False
```

Comment:
`self._set_backend_pq` should be kept. 
The current design is a little bit wrong I believe. Since loading PQ in the backend also needs to compute the codebook distance, and if the PQ is untrained at the init stage, the codebook distance in the backend is meaningless(computed from zeros). 
So only when the `self.pq_codec` is not `None`  and is `trained`, we then need to load it into backend. This part of logic is in the `pre_process`, which was deleted.